tests/testsuite/credential_process.rs

```diff
@@ -32,7 +32,7 @@ fn gated() {
         .build();
 
     p.cargo("publish --no-verify")
-        .masquerade_as_nightly_cargo()
+        .masquerade_as_nightly_cargo(&["credential-process"])
```

Comment:
Something is non-obvious here since the `-Zcredential-process`  is unspecified for this and the next test

Reply:
I'll dig a bit but removing `masquerade_as_nightly_cargo` allows it to work but the test is called `gated`. Maybe it should have `-Zcredential-process`?

Reply:
I updated everything but this. I honestly have no idea why `masquerade_as_nightly_cargo` is here.

 @ehuss it looks like you wrote this, do you have any ideas?

Reply:
There's not really a strong reason to have it one way or the other. Switching to nightly mode makes extra sure that it does not implicitly use the feature without the `-Z` flag. There are some things that behave differently in nightly mode, and I was probably just trying to be extra paranoid here.  But it is also unlikely for that to be a problem in practice.